common/src/main/java/com/wynntils/core/config/ConfigHolder.java

```diff
@@ -103,27 +103,39 @@ private String getDeclaringFeatureNameCamelCase() {
         return CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_CAMEL, name);
     }
 
-    public Config getMetadata() {
-        return metadata;
+    public String getSubcategory() {
+        return subcategory;
+    }
+
+    public String getI18nKey() {
+        return i18nKey;
+    }
+
+    public boolean isVisible() {
+        return visible;
     }
 
     public String getDisplayName() {
-        if (!getMetadata().key().isEmpty()) {
-            return I18n.get(getMetadata().key() + ".name");
+        if (!getI18nKey().isEmpty()) {
+            return I18n.get(getI18nKey() + ".name");
         }
-        return ((Translatable) parent).getTranslation(field.getName() + ".name");
+        return ((Translatable) parent).getTranslation(getFieldName() + ".name");
     }
 
     public String getDescription() {
-        if (!getMetadata().key().isEmpty()) {
-            return I18n.get(getMetadata().key() + ".description");
+        if (!getI18nKey().isEmpty()) {
+            return I18n.get(getI18nKey() + ".description");
         }
-        return ((Translatable) parent).getTranslation(field.getName() + ".description");
+        return ((Translatable) parent).getTranslation(getFieldName() + ".description");
     }
 
     public Object getValue() {
+        return getConfig().get();
+    }
+
+    private Config getConfig() {
         try {
-            return FieldUtils.readField(field, parent, true);
+            return (Config) FieldUtils.readField(field, parent, true);
```

Comment:
Why do we need any reflection at all when we have the wrapper? Couldn't we convert all of this to not use reflection (other than for discovering configs)?

Reply:
We probably can. As I said:

> Further cleanup of ConfigManager and unification with Storage is left as a future exercise; this is just the minimal amount of work needed to change all annotation to wrapper classes.

Reply:
This change is scary enough as it is. I don't want to do any changes that are not strictly necessary.